### PR TITLE
an: new recipe

### PR DESCRIPTION
**Pull request by lecast0r**
I found this interesting little program on the Debian project's package website and decided to port it to Haiku.

While I've tested this on x86_64 within VirtualBox, I have not yet tested with x86_gcc2.

P.S. This is my first PR ever, not to mention my first Haiku recipe, let's hope I didn't screw something up :)

### CODE REVIEW

**Comment by waddlesplash** [comment]
Why did you close this? It may need some minor comments but nothing too drastic...

**Comment by lecast0r** [comment]
> It may need some comments but nothing too drastic...

Comments in the recipe file? 

As for why I closed it, the automated build failure was outside my "realm" of expertise and I wouldn't know what to do with it. It was involving the "lint-new-recipes.sh" script, whatever that does.

**Comment by waddlesplash** [comment]
Yes, and if you want to learn, we can help you with that...

The script probably gave an error message in the log as to what was wrong. What was it?

**Comment by Begasus** [comment]
Builds OK on 32bit Haiku (with gcc2), made some local changes, not sure how to use it but it seems to be working OK :)

**Comment by Begasus** [comment]
Pushed some changes based on your work to https://github.com/Begasus/haikuports/commit/187ee29e20ec1b890648b65ee70b652a3c1d638b ...

File: games-misc/an/an-1.2.recipe [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+DESCRIPTION="an is a fast anagram generator that uses the system's built-in \
+word dictionary to form anagrams. an supports Unicode and theoretically should \
+work for any alphabetic language."
+HOMEPAGE="https://salsa.debian.org/pm/an"
+COPYRIGHT="2012-2014 Paul Martin"
+LICENSE="GNU GPL v2"
+REVISION="1"
+# cloning through git doesn't seem to work nor does downloading an archive
+# from salsa.debian.org, so use the Debian archives instead
+#SOURCE_URI="https://salsa.debian.org/pm/-/archive/master/an-master.tar.gz"
```
Comment (Begasus) [inline games-misc/an/an-1.2.recipe:11]: This could do the trick: https://salsa.debian.org/pm/an/-/archive/403d9a4fd75a5cfae9d5388c4a0e7f889145177f.tar.gz

**Comment by lecast0r** [comment]
> The script probably gave an error message in the log as to what was wrong. What was it?

I found the automated build log from TravisCI, however, the error message didn't mention anything that might be of use.

The main thing that stands out is `The command "./lint-new-recipes.sh --config=haikuports.conf --licenses=haiku-licenses-master" exited with 1.` I'm wondering if the whitespace that the log below mentioned was inconsistent somehow, either in the recipe file or one of the patches files.

[log.txt](https://github.com/haikuports/haikuports/files/4531104/log.txt)

Regarding the changes that @Begasus had made, I looked at them, and noticed he/she had changed some of the `REQUIRES` and `BUILD_REQUIRES` statements. I had tried using `lib:icu` and `devel:icu_devel` at first, thinking that the package names would be the correct form, but apparently they weren't. Maybe that's part of what was causing the build errors too.

**Comment by Begasus** [comment]
> > The script probably gave an error message in the log as to what was wrong. What was it?
> 
> I found the automated build log from TravisCI, however, the error message didn't mention anything that might be of use.
> 
> The main thing that stands out is `The command "./lint-new-recipes.sh --config=haikuports.conf --licenses=haiku-licenses-master" exited with 1.` I'm wondering if the whitespace that the log below mentioned was inconsistent somehow, either in the recipe file or one of the patches files.
> 

Main error with Travis was the failing whitespace

> [log.txt](https://github.com/haikuports/haikuports/files/4531104/log.txt)
> 
> Regarding the changes that @Begasus had made, I looked at them, and noticed he/she had changed some of the `REQUIRES` and `BUILD_REQUIRES` statements. I had tried using `lib:icu` and `devel:icu_devel` at first, thinking that the package names would be the correct form, but apparently they weren't. Maybe that's part of what was causing the build errors too.

If you look at the actual build process you see those 3 icu libraries being used, if you check the PROVIDES in the recipe for icu you can find the mentioned libraries there: https://github.com/haikuports/haikuports/blob/48532120bc499a4513397153543c0e40734d3073/dev-libs/icu/icu-57.2.recipe#L62

**Comment by Begasus** [comment]
PS, me = he :)

File: games-misc/an/an-1.2.recipe [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+SOURCE_FILENAME="an-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="an-$srcGitRev"
+
+ARCHITECTURES="x86_gcc2 x86_64"
+
+PROVIDES="
+	$portName = $portVersion
+	cmd:an = $portVersion
+	"
+
```
Comment (Begasus) [inline games-misc/an/an-1.2.recipe:21]: No new line here

File: games-misc/an/an-1.2.recipe [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	lib:libicudata
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libicui18n
+	devel:libicuuc
+	devel:libicudata
+	"
+
```
Comment (Begasus) [inline games-misc/an/an-1.2.recipe:35]: No new line here

**Comment by Begasus** [comment]
> This should be in app-text, haikuports follows gentoo-portage when an app also exists there, and this one is in app-text there.
> https://packages.gentoo.org/categories/app-text

Thanks @scottmc , while you move it @lecast0r could you rename the recipe to **an-1.2~git.recipe** (as this is a commit checkout)

**Comment by lecast0r** [comment]
> > This should be in app-text, haikuports follows gentoo-portage when an app also exists there, and this one is in app-text there.
> > https://packages.gentoo.org/categories/app-text
> 
> Thanks @scottmc , while you move it @lecast0r could you rename the recipe to **an-1.2~git.recipe** (as this is a commit checkout)

Absolutely. I guess the reason i mistakingly classified it as a game was because under Debian it was classified as such, and my search of the Gentoo packages list yielded packages that had "an" in their name or description, so my laziness told me to not search for it any further because there were so many results.

In short, my laziness is likely to blame for all the problems thus far 🙃

**Comment by Begasus** [comment]
Looking at https://repology.org/project/an/versions it's not so hard to use games-misc :) but yes, we follow gentoo's layout where we can :)

**Comment by Begasus** [comment]
I'm not a git expert, but something went wrong here :)

**Comment by lecast0r** [comment]
> I'm not a git expert, but something went wrong here :)

Neither am I, but all I do know is that I accidentally removed the good recipe file >:(
Fortunately since commit history exists i was able to restore it.

**Comment by Begasus** [comment]
You merged the new changes from master into your branch it seems ...

**Comment by lecast0r** [comment]
> You merged the new changes from master into your branch it seems ...

Sure did. I'm not even sure if anything will work now. But it seems the Travis CI build worked this time...

**Comment by Begasus** [comment]
You have 3 unwanted (changed) files in here now :)

**Comment by lecast0r** [comment]
> You have 3 unwanted (changed) files in here now :)

Hmm... should I say which is the "correct" one, or is there something else I should do?

**Comment by Begasus** [comment]
> > You have 3 unwanted (changed) files in here now :)
> 
> Hmm... should I say which is the "correct" one, or is there something else I should do?

If you look at this PR in a browser you'll see 4 files added, we only need the one for this request :)

**Comment by Begasus** [comment]
Deleting the 3 files doesn't solve the problem, maybe someone with more knowledge on git could step in here?

**Pull request merged**